packages/components/react/package.json

```diff
@@ -8,11 +8,31 @@
   "license": "MIT",
   "types": "./dist/index.d.ts",
   "exports": {
-    ".": "./dist/index.js"
+    ".": "./dist/index.js",
+    "./core": "./dist/core.js",
+    "./core/CodeMirrorEditor": "./dist/core/CodeMirrorEditor/index.js",
+    "./core/FileTree": "./dist/core/FileTree.js",
+    "./core/Terminal": "./dist/core/Terminal/index.js"
   },
   "files": [
     "dist"
   ],
+  "typesVersions": {
+    "*": {
+      "core": [
+        "dist/core.d.ts"
+      ],
+      "./core/CodeMirrorEditor": [
+        "dist/core/CodeMirrorEditor/index.d.ts"
+      ],
+      "./core/FileTree": [
+        "dist/core/FileTree.d.ts"
+      ],
+      "./core/Terminal": [
+        "dist/core/Terminal/index.d.ts"
+      ]
+    }
+  },
```

Comment:
We could declare these in the `exports` directly, e.g.

```json
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    },
    "./core": {
      "import": {
        "types": "./dist/core.d.ts",
        "default": "./dist/core.js"
      }
    },
    "./core/CodeMirrorEditor": {
      "types": "./dist/core/CodeMirrorEditor/index.d.ts",
      "default": "./dist/core/CodeMirrorEditor/index.js"
    },
```

Reply:
Oh this is nice! I'll give it a go :raised_hands: